### DISCUSSION TITLE
Proposal for new `simpleafR` package — loading and *basic* pre-processing of `alevin-fry` output in R

### DISCUSSION THREAD

**Discussion by rob-p** [COMBINE-lab/alevin-fry#117] [Ideas]
This discussion is meant to follow up on the closing post of @wmacnair in #55 (cc @DongzeHE, @mikelove).  The relevant part of that post is reproduced below:

However, reading through again it was clear to me that it would be great to have a small, purpose-built standalone R package for loading simpleaf / alevin-fry outputs into R.

At the moment you suggest using fishpond::loadFry. This works fine, but I find it a bit unsatisfying, partly because it's not the main purpose of the package, but also some additional functionality would be great.

For example, I would love to see a package, e.g. simpleafR, that does:

* loading processed data from multiple directories in one function call
* excluding barcodes with zero reads
* produce either sce or Seurat objects as outputs
* (perhaps an option would be to combine this with alevinQC?)

I'd like to gauge interest in this, as well as get feedback on:

1. What the most wanted / useful functionality would be.
2. If anyone from the community would be interested in contributing to such an effort.

**Comment by wmacnair** [comment]
Hi @rob-p, thanks for starting this discussion :)

Some arguments in favour of a standalone package:
- The ideal would be for the whole process to be as low effort as possible for the average user to get from FASTQ to `sce` / `Seurat` / `anndata`. `simpleaf` did this for the mapping stage; it's still a bit more effort than with `CellRanger` for the last step.
- It makes it super obvious to new users what they need to use.
- `fishpond::loadFry` is fine, but it doesn't have the functionality of `DropletUtils::read10xCounts`.
- The bulk of `fishpond` the package is only tangentially related to the very common task of "load my counts into R".

I think you could also achieve "obviousness" by adding a `readAlevinCounts` function to `DropletUtils`. Although I think that `DropletUtils` also has the "many functions not used by the typical user" issue that I mentioned with `fishpond`.

I don't think I would have time to properly contribute, but I'd be happy to help with testing / giving feedback on documentation 😄 

Cheers
Will

**Comment by wmacnair** [comment]
Ah and a couple of thoughts on possible functionality:
- functionality to import multimodal data
- minor, but I think including _total_spliced_ / _total_unspliced_ (perhaps more accurately _total_spliced_ / _total_intronic_, if I've understood `splici` vs `spliceu` correctly?) as cell-level QC metrics would be great and should be encouraged :D

**Comment by mikelove** [comment]
Re:fishpond. Agree not everything has to go there. Only if it’s useful for the developer.

 My notes from the other thread:

We have some stuff in fishpond for import and processing, and welcome to consider other things to go in there. Just need docs and tests really. There’s even some new stuff for working with EC counts. But it doesn’t have to go there if you want a stand-alone functionality.

Recently we took out non-R code to move to a separate package for ease of installation. EDS package on Bioc (Avi’s) now has the original alevin data storage format which is read in with C++. 

So that’s one consideration: we don’t want extra dependency burden for fishpond.

**Comment by rob-p** [comment]
Thanks @wmacnair for the overview and suggestions and @mikelove for the thoughts wrt fishpond.  I agree that a lot of the functionality described above doesn't make sense to live in fishpond itself.  However, I wonder if there is an argument for a new package `simpleafR` (probably we should think of a different name) that has `fishpond` as a dependency.

That is, I'd also like to avoid spreading out / re-writing the R infrastructure for handling the `alevin-fry` directory format and reading in the USA matrices.  So, a higher level package could depend on `fishpond` for doing the import/export and some basic data munging, and the could itself implement some of the higher-level functionality that Will discuses.

Any thoughts on a package like that?

**Comment by mikelove** [comment]
Briefly: if you want you can also have a package as a Suggests, which means it is not required for installation but is used for testing. You can therefore have users prompted to install an extra package only when needed.
